bouncerscript/test/test_script.py

```diff
@@ -39,13 +40,28 @@ def test_main(submission_context, event_loop, fake_session):
 # bouncer_submission {{{1
 @pytest.mark.asyncio
 async def test_bouncer_submission(submission_context, mocker):
-    mocker.patch.object(bscript, 'does_product_exists', new=noop_async)
+    mocker.patch.object(bscript, 'does_product_exist', new=noop_async)
     mocker.patch.object(bscript, 'api_add_product', new=noop_async)
     mocker.patch.object(bscript, 'api_add_location', new=noop_async)
+    mocker.patch.object(bscript, 'api_show_location', new=noop_async)
+    with pytest.raises(ScriptWorkerTaskException):
+        await bouncer_submission(submission_context)
+
+    mocker.patch.object(bscript, 'does_product_exist', new=toggled_boolean_async)
+    mocker.patch.object(bscript, 'check_locations_match', new=return_false_sync)
+    mocker.patch.object(bscript, 'check_path_matches_destination', new=return_true_sync)
+    with pytest.raises(ScriptWorkerTaskException):
+        await bouncer_submission(submission_context)
+
+    mocker.patch.object(bscript, 'does_product_exist', new=toggled_boolean_async)
```

Comment:
Hm, how does this work? Does re-mocking give you a reset call count? Aiui the mocker mock exists until the test finishes.

Reply:
Yeah, but you're right, it's not needed. I just tested with the following two occurrences removed and it works smoothly. The way the test is written, it's fine, since it goes up with +2 for each call of that function so it's like a reset.